src/unit.c

```diff
@@ -166,78 +165,65 @@ int setCycleTime(char *input, char *sendBuf)
     return 8;
 }
 
+int bcd2dec(int bcd) {
+    int dec = bcd - ((int)(bcd / 16) * 6);
+    return dec;
+}
+
 int getSysTime(char *recv, int len, char *result)
 {
-    char day[3];
+    struct tm t = {0};
 
     if (len != 8) {
         sprintf(result, "System time: Len <> 8 bytes");
         return 0;
     }
 
-    // TODO: Can this be translated, or is it used as German day abbreviations
-    // when communicating with the heating controller?
-    switch (recv[4]) {
-    case 0:
-        strcpy(day, "So");
-        break;
-    case 1:
-        strcpy(day, "Mo");
-        break;
-    case 2:
-        strcpy(day, "Di");
-        break;
-    case 3:
-        strcpy(day, "Mi");
-        break;
-    case 4:
-        strcpy(day, "Do");
-        break;
-    case 5:
-        strcpy(day, "Fr");
-        break;
-    case 6:
-        strcpy(day, "Sa");
-        break;
-    case 7:
-        strcpy(day, "So");
-        break;
-    default:
-        sprintf(result, "Error day conversion: %02X", recv[4]);
-        return 0;
-    }
+    t.tm_year = bcd2dec(recv[0]) * 100 + bcd2dec(recv[1]) - 1900;
+    t.tm_mon = bcd2dec(recv[2]) - 1;
+    t.tm_mday = bcd2dec(recv[3]);
+    t.tm_wday = bcd2dec(recv[4]) % 7;
+    t.tm_hour = bcd2dec(recv[5]);
+    t.tm_min = bcd2dec(recv[6]);
+    t.tm_sec = bcd2dec(recv[7]);
 
-    sprintf(result, "%s,%02X.%02X.%02X%02X %02x:%02X:%02X", day,
-            recv[3], recv[2], recv[0], recv[1],
-            recv[5], recv[6], recv[7]);
+    // This might break existing applications. But changing the string to some custom format
+    // that is not recognized by strptime for parsing dates has a symmetry impact that the
+    // format for getTime is different from the format required by setTime.
+    // I would consider the command symmetry more important for the future.
+    strftime(result, 48, "%c", &t);
```

Comment:
I'd prefer this to use [ISO 8601](https://de.wikipedia.org/wiki/ISO_8601) instead of some locale specific format.